### ISSUE TITLE
Holster Anim missing on Prop Pickup (+use)

### ISSUE THREAD

**Issue by burntfrenchfries** [Facepunch/garrysmod-issues#5911]
Running on x86-64 branch, pretty simple issue.

In HL2, on picking a prop, your gun slowly goes down in a holstering anim, in GMoD it just disappears, with lack of any animation. Wondering if this is an unintended issue?

**Comment by ValsdalV** [comment]
I know this issue is old, sorry, but I have some additional info to help with debugging.

It appears that the holstering animation plays correctly when having a `weapon_annabelle` or `weapon_alyxgun` equipped; these weapons share their view models with `weapon_shotgun` (`models/weapons/c_shotgun.mdl`) and `weapon_pistol` (`models/weapons/c_pistol.mdl`), respectively.

Also, picking up and dropping down the same prop repeatedly, with the Annabelle or Alyx Gun equipped, leads to view model hands disappearing.

**Comment by Plu1223** [comment]
>  Also, picking up and dropping down the same prop repeatedly, with the Annabelle or Alyx Gun equipped, leads to view model hands disappearing.

This issue is actually still present with every other weapon in the game, however, it must be done really quickly in order for it to happen. I believe it's easier to preform on those weapons since they have a short time of still being visible before going invisible unlike the rest of the weapons that immediately disappear.

**Comment by robotboy655** [comment]
I have looked into it, the animations are not missing, it's just that HL2DM weapons hide the weapon on holster even if the animation plays. It can be disabled, but the other issue should be fixed first, regarding the disappearing hands.

I have also looked into the disappearing hands, but it's a bit more complex, it's some networking issue where the hands are being unparented from the viewmodel and never reparented. Don't have a fix for this yet.

**Comment by PhoenX279** [comment]
I think I figured out the cause for the hands to not appear.

So basically, while testing out a SWEP I made, I found out the hard way that picking up a prop and dropping it instantly counts as a proper "holster" but doesn't count as a proper "draw/deploy." Because of this, whatever lines of code that would get executed in "function SWEP:Deploy()" get skipped.

And I think the same thing is happening with normal weapons here, though I'm not 100% sure.
Basically just like how lines in "SWEP:Deploy()" gets skipped, I believe that the line responsible for adding c_hands to viewmodels also gets skipped, resulting in invisible hands.

This happens on every default weapon that supports c_hands, not just `annabelle` and `alyxgun`
